### ISSUE TITLE
Property "hibernate.javax.cache.uri" is ignored in JakartaEE deployments

### ISSUE THREAD

**Issue by mletenay** [ben-manes/caffeine#1347]
**Situation:**
JakartaEE application with EAR file consiting of several EJB JAR modules, each with its own persistence context and its own hibernate 2nd level cache config.

**Problem:**
Migrating from EhCache 2.x to Caffeine, but as of 3.1.8 it does not work and completely ignores the _hibernate.javax.cache.uri_ specified in _persistence.xml_ descriptors.
The issue with "hibernate.javax.cache.uri" was already partially addressed in https://github.com/ben-manes/caffeine/issues/877 but that solution works only for Spring application.

As mention in https://github.com/ben-manes/caffeine/issues/877#issuecomment-1454981294 from security reasons, the uri loading was limited to **file:** and **classpath:** protocols.
The actual trouble is that in case of JakartaEE, the URI is neither of those but something like **jar:**file:/ejbmodule.jar!/META-INF/specific-cache.conf.

The fix is be trivial, just add jar: protocol handling to TypesafeConfigurator#resolveConfig, I will create PR for that ...

**Comment by ben-manes** [comment]
Thank you! I’m hoping to get caught up over the weekend for this and the other two recent requests. Until this is released, you can use [setConfigSource](https://github.com/ben-manes/caffeine/blob/41cf7fdf7c0f7f62c502467f31d08cce0e43814b/jcache/src/main/java/com/github/benmanes/caffeine/jcache/configuration/TypesafeConfigurator.java#L151) to resolve using your application strategy.

**Comment by ben-manes** [comment]
I merged this a little early to base the unit test on top of. It took a little build magic to make an extra test resource jar that is placed on the classpath, so that the resource lookup resolves using the jar scheme. That uri is working so just need to duplicate the tests cases. I might get everything wrapped up this weekend for a release, but no promises.

**Comment by mletenay** [comment]
I've looked at the unit tests when pushed my commit, but I am more junit/maven guy and your setup is slightly above my skills...
Anyway, thanks for nice library.

**Comment by mletenay** [comment]
Sorry to bother you, but it would be nice to have a new release so we do not have to hack around this issue with custom call of seeConfigSource ...
Thanks a lot.

**Comment by ben-manes** [comment]
sorry, I wasn't as active recently and spent time investigating recent claims that our eviction policy was under performing when compared to new algorithms from CMU. It seems they made very minor, mostly inconsequential changes to old algorithms and the issue they observed was with their attempt to reimplement our algorithms. Caffeine aced their workload traces whereas their algorithms did poorly on the wider set of traces that I use. I'll try to get back to the outstanding issues to unblock a release.

**Comment by ben-manes** [comment]
Released in 3.2.0 (sorry for the delay)
